study-builder/studies/circadia/emails/consent_received.tmpl.html

```diff
@@ -0,0 +1,144 @@
+<!DOCTYPE html>
+
+<html>
+<head>
+  <meta
+    http-equiv="Content-Type"
+    content="text/html; charset=utf-8"
+  />
+  <meta
+    name="viewport"
+    content="width=device-width, initial-scale=1, minimum-scale=1, maximum-scale=1"
+  />
+
+  <link
+    rel="preconnect"
+    href="https://fonts.gstatic.com"
+  />
+  <link
+    rel="stylesheet"
+    href="https://fonts.googleapis.com/css2?family=Nunito:wght@300;400;600;700&display=swap"
+  />
+
+  <!--[if !mso]><!-->
+  <meta
+    http-equiv="X-UA-Compatible"
+    content="IE=Edge"
+  />
+  <title>
+      ${text["subject"]}
+  </title>
+  <!--<![endif]-->
+
+  <!--[if (gte mso 9)|(IE)]>
+  <xml>
+    <o:OfficeDocumentSettings>
+      <o:AllowPNG/>
+      <o:PixelsPerInch>96</o:PixelsPerInch>
+    </o:OfficeDocumentSettings>
+  </xml>
+  <![endif]-->
+
+  <!--[if (gte mso 9)|(IE)]>
+  <style type="text/css">
+        body {
+          width: 600px;
+          margin: 0 auto;
+        }
+
+        table {
+          border-collapse: collapse;
+        }
+
+        table,
+        td {
+          mso-table-lspace: 0pt;
+          mso-table-rspace: 0pt;
+        }
+
+        img {
+          -ms-interpolation-mode: bicubic;
+        }
+  </style>
+  <![endif]-->
+
+  <style type="text/css">
+      body,
+      p,
+      div {
+        font-family: Nunito, sans-serif;
+        font-size: 16px;
+        color: #282828;
+      }
+
+      p {
+        margin: 0 0 20px 0;
+        padding: 0;
+      }
+  </style>
+</head>
+<body>
+<center>
+  <table
+    style="
+      width: 100% !important;
+      max-width: 600px !important;
+      border-collapse: collapse !important;
+      background-color: #ffffff !important;
+    "
+  >
+    <tbody>
+    <tr>
+      <td style="text-align: center; padding: 20px">
+        <img
+          src="https://storage.googleapis.com/${assetsBucketName}/auth0/logo.png"
```

Comment:
Is this a correct way to use available variables in email templates?

Reply:
Yup! There's 2 levels of templating here. The first is [Apache Velocity](https://velocity.apache.org/engine/1.7/user-guide.html) templating which uses the `${...}` syntax (and is enabled by the `render` flag in `sendgrid-emails.conf`. Next is sendgrid's handlebars, which uses `{{...}}` syntax. study-builder renders the Apache Velocity ones when creating the email, then sendgrid renders the other one when the email is actually sent during runtime. More details in the wiki here: https://broadinstitute.atlassian.net/wiki/spaces/DDP/pages/1182630029/Emails+Guide.